### ISSUE TITLE
Drafting a PoC for the lightning LSP

### ISSUE THREAD

**Issue by vincenzopalazzo** [vincenzopalazzo/lampo.rs#254]
We should try to implement by the next release a PoC that allows running the 
LSP using the ldk crate https://github.com/lightningdevkit/lightning-liquidity

**Comment by vincenzopalazzo** [comment]
Now ldk included https://github.com/lightningdevkit/lightning-liquidity inside the repo but lampo is not updated with the last ldk version

So we should work on the one that @Harshit933 was working on in the following PR https://github.com/vincenzopalazzo/lampo.rs/pull/277

So the best approach is to take the last version of lampo from the `next` branch and include the changes that are https://github.com/vincenzopalazzo/lampo.rs/pull/292

Then implement the server side in a new crate like `lampo-lsp-server` and then use ldk-node as a client to see if the server is able to work.

**Comment by Harshit933** [comment]
Agreed we should take this step by step, first implementing as a server and testing it. Then we should move towards adding other things like taking Liquidity from other sources.
